plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java

```diff
@@ -47,8 +50,28 @@ public ImportApplicationWizard() {
 		this.model = new ImportApplicationWizardModel();
 	}
 
+	public ImportApplicationWizard(Map<IProject, Collection<IBuildConfig>> projectsAndBuildConfigs) {
+		this();
+		if (projectsAndBuildConfigs != null && projectsAndBuildConfigs.size() == 1) {
+			Map.Entry<IProject, Collection<IBuildConfig>> entry = projectsAndBuildConfigs.entrySet().iterator().next();
+			IProject project = entry.getKey();
+			Connection connection = ConnectionsRegistryUtil.safeGetConnectionFor(project);
+			setModelConnection(connection);
+			Collection<IBuildConfig> buildConfigs = entry.getValue();
+			if (buildConfigs.size() == 1) {
+				model.setSelectedItem(buildConfigs.iterator().next());
+			} else {
+				model.setSelectedItem(project);
```

Comment:
if 1 buildconfig is found, then we skip the buildconfig selection page and go directly to the git clone page.
If more buildconfigs are found, we select the parent project and display the buildconfig selection page (see addPages()). The user will have to choose what to import